pkg/smokescreen/smokescreen.go

```diff
@@ -314,9 +318,20 @@ func dialContext(ctx context.Context, network, addr string) (net.Conn, error) {
 	}
 	sctx.logger = sctx.logger.WithFields(fields)
 
-	// Only wrap CONNECT conns with an InstrumentedConn. Connections used for traditional HTTP proxy
+	// Only wrap CONNECT conns and MITM http conns with an InstrumentedConn. Connections used for traditional HTTP proxy
 	// requests are pooled and reused by net.Transport.
-	if sctx.proxyType == connectProxy {
+	if sctx.proxyType == connectProxy || pctx.ConnectAction == goproxy.ConnectMitm {
+		// If we have a MITM and option is enabled, we can add detailed Request log fields
```

Comment:
Could we shift this up out of this scope since this is mostly related to creating the `InstrumentedConn`? These checks could be tossed into a function and extend the logger above.

Reply:
Great idea. Moved all the logger fields to a `dialContextLoggerFields` func